### PR TITLE
Sélection d'exemples de produits textile

### PR DESCRIPTION
**Pull request by magopian**
[user story](https://www.notion.so/fd0068a7db8348c584cd53a3eea399f2?v=af2332265058414da4bd3aa4d381ed5f&p=5cec5bca85794d37b39602650601eb04&pm=s)

### CODE REVIEW

File: src/Data/Textile/Product.elm [hunk 1]
```diff
@@ -97,6 +97,11 @@ getFabricProcess maybeKnittingProcess { fabric } wellknown =
             process
 
 
+getFadingByDefault : Product -> Bool
+getFadingByDefault product =
+    product.id == Id "jean"
```
Comment (n1k0) [inline src/Data/Textile/Product.elm:102]: May I suggest `isFadedByDefault` instead?

File: src/Data/Textile/Simulator.elm [hunk 2]
```diff
@@ -222,7 +222,7 @@ computeMakingImpacts { wellKnown } ({ inputs } as simulator) =
                                 { makingComplexity = inputs.makingComplexity |> Maybe.withDefault inputs.product.making.complexity
                                 , fadingProcess =
                                     -- Note: in the future, we may have distinct fading processes per countries
-                                    if inputs.product.making.fadable && inputs.disabledFading /= Just True then
+                                    if inputs.disabledFading == Just False || (inputs.disabledFading == Nothing && Product.getFadingByDefault inputs.product) then
```
Comment (n1k0) [inline src/Data/Textile/Simulator.elm:225]: May I suggest introducing a new `isFaded` function in the `Inputs` module, so we can simply write `if Product.isFaded inputs then`?

File: src/Page/Textile/Simulator.elm [hunk 3]
```diff
@@ -570,25 +609,27 @@ massField massInput =
...
+            [ text "Produit" ]
+        , button
+            [ class "form-select ElementSelector text-start"
+            , id "selector-example"
+            , onClick (SetModal (SelectExampleModal autocompleteState))
+            ]
+            [ span
+                []
+                [ text <| Inputs.exampleProductToString query ]
+            ]
```
Comment (n1k0) [inline src/Page/Textile/Simulator.elm:632]: is this empty span actually needed?

File: src/Views/Textile/Step.elm [hunk 4]
```diff
@@ -324,15 +324,23 @@ fadingField { inputs, toggleDisabledFading } =
...
+                (inputs.disabledFading
+                    |> Maybe.map not
+                    |> Maybe.withDefault (Product.getFadingByDefault inputs.product)
+                )
             , onCheck (\checked -> toggleDisabledFading (not checked))
             ]
             []
-        , if inputs.disabledFading == Just True then
-            text "Délavage désactivé"
+        , case ( inputs.disabledFading, Product.getFadingByDefault inputs.product ) of
```
Comment (n1k0) [inline src/Views/Textile/Step.elm:335]: That's also a place `Inputs.isFaded inputs` might come useful and more legible

**Review by n1k0 — APPROVED** [review]
Looks great, just a few nits.

Also, I agree `disabledFading` should actually be `fading=true/false`, but this should be addressed in another PR if POs agree.

**Comment by magopian** [comment]
Nice, thanks for the valuable feedback, added a commit, if that works for you I'll land once the checks are green 👍

**Comment by magopian** [comment]
Oh, wait, forgot to add the base examples first

**Comment by magopian** [comment]
Ok, added a bunch of examples, if @n1k0 you want to give it a last look?

**Review by n1k0 — APPROVED** [review]
@magopian very nice! ship it 🛳️

**Pull request merged**